parser/wasm_parse/mod.rs

```diff
@@ -589,31 +707,38 @@ impl<'a> Parse<'a> for wasmparser::TypeSectionReader<'a> {
             let (ty, size) = ty?;
             let id = Id::entry(idx, i);
 
-            let mut name = format!("type[{}]: (", i);
-            for (i, param) in ty.params.iter().enumerate() {
-                if i != 0 {
-                    name.push_str(", ");
-                }
-                name.push_str(ty2str(*param));
-            }
-            name.push_str(") -> ");
-
-            match ty.returns.len() {
-                0 => name.push_str("nil"),
-                1 => name.push_str(ty2str(ty.returns[0])),
-                _ => {
-                    name.push_str("(");
-                    for (i, result) in ty.returns.iter().enumerate() {
+            match ty {
+                wasmparser::TypeDef::Func(func) => {
+                    let mut name = format!("type[{}]: (", i);
+                    for (i, param) in func.params.iter().enumerate() {
                         if i != 0 {
                             name.push_str(", ");
                         }
-                        name.push_str(ty2str(*result));
+                        name.push_str(ty2str(*param));
+                    }
+                    name.push_str(") -> ");
+
+                    match func.returns.len() {
+                        0 => name.push_str("nil"),
+                        1 => name.push_str(ty2str(func.returns[0])),
+                        _ => {
+                            name.push_str("(");
+                            for (i, result) in func.returns.iter().enumerate() {
+                                if i != 0 {
+                                    name.push_str(", ");
+                                }
+                                name.push_str(ty2str(*result));
+                            }
+                            name.push_str(")");
+                        }
                     }
-                    name.push_str(")");
+
+                    items.add_item(ir::Item::new(id, name, size, ir::Misc::new()));
                 }
+                // @@@ Is this correct?
```

Comment:
To be honest I don't know if this is correct, but it looks like it preserves the previous behavior so i guess yes :D